sdk/python/kubeflow/tfjob/api/tf_job_client.py

```diff
@@ -149,3 +175,115 @@ def delete(self, name, namespace=None):
+      status_callback: (Optional): Callable. If supplied this callable is
+        invoked after we poll the job. Callable takes a single argument which
+        is the job.
+    """
+    if namespace is None:
+      namespace = utils.get_default_target_namespace()
+
+    return self.wait_for_condition(
+      name,
+      ["Succeeded", "Failed"],
+      namespace=namespace,
+      timeout_seconds=timeout_seconds,
+      polling_interval=polling_interval,
+      status_callback=status_callback)
+
+
+  def wait_for_condition(self, name,
+                         expected_condition,
+                         namespace=None,
+                         timeout_seconds=600,
+                         polling_interval=30,
+                         status_callback=None):
+    """Waits until any of the specified conditions occur.
+
+    Args:
+      name: Name of the job.
+      expected_condition: A list of conditions. Function waits until any of the
+        supplied conditions is reached.
+      namespace: defaults to current or default namespace.
+      timeout_seconds: How long to wait for the job.
+      polling_interval: How often to poll for the status of the job.
+      status_callback: (Optional): Callable. If supplied this callable is
+        invoked after we poll the job. Callable takes a single argument which
+        is the job.
+    """
+
+    if namespace is None:
+      namespace = utils.get_default_target_namespace()
+
+    for _ in range(round(timeout_seconds/polling_interval)):
+
+      tfjob = None
+      tfjob = self.get(name, namespace=namespace)
+
+      if tfjob:
+        if status_callback:
+          status_callback(tfjob)
+
+        # If we poll the CRD quick enough status won't have been set yet.
+        conditions = tfjob.get("status", {}).get("conditions", [])
+        # Conditions might have a value of None in status.
+        conditions = conditions or []
+        for c in conditions:
+          if c.get("type", "") in expected_condition:
+            return tfjob
+
+      time.sleep(polling_interval)
+
+    raise RuntimeError(
+      "Timeout waiting for TFJob {0} in namespace {1} to enter one of the "
+      "conditions {2}.".format(name, namespace, expected_condition), tfjob)
+
+
+  def get_job_status(self, name, namespace=None):
+    """Returns TFJob status, such as Running, Failed or Succeeded.
+
+    Args:
+      name: The TFJob name.
+      namespace: defaults to current or default namespace.
+    """
+    if namespace is None:
+      namespace = utils.get_default_target_namespace()
+
+    tfjob = self.get(name, namespace=namespace)
+    last_condition = tfjob.get("status", {}).get("conditions", [])[-1]
+    return last_condition.get("type", "")
+
+
+  def if_job_running(self, name, namespace=None):
+    """Returns true if the TFJob running; false otherwise.
+
+    Args:
+      name: The TFJob name.
+      namespace: defaults to current or default namespace.
```

Comment:
current or default?

Reply:
Thanks Ce! Yes, will get the current namespace if running in the cluster from `/var/run/secrets/kubernetes.io/serviceaccount/namespace`, otherwise will get the default namespace. please see below for the logic, thanks.

https://github.com/kubeflow/tf-operator/blob/0267710c69f2f92e4a65db597f7ee0d2e734462f/sdk/python/kubeflow/tfjob/utils/utils.py#L26

Reply:
SGTM.